py4cast/datasets/access.py

```diff
@@ -0,0 +1,435 @@
+
+    @property
+    def terms_list(self) -> np.array:
+        return np.arange(self.term_start, self.term_end + 1, self.step_duration)
+
+    @property
+    def date_list(self) -> np.array:
+        """
+        List all dates available for the period, with a 24h leap
+        """
+        return np.arange(
+            self.start,
+            self.end + np.timedelta64(1, "D"),
+            np.timedelta64(1, "D"),
+            dtype="datetime64[s]",
+        ).tolist()
+
+
+@dataclass
+class Timestamps:
+    """
+    Describe all timestamps in a sample.
+    It contains
+        datetime, terms, validity times
+
+    If n_inputs = 2, n_preds = 2, terms will be (-1, 0, 1, 2) * step_duration
+     where step_duration is typically an integer multiple of 1 hour
+
+    validity times correspond to the addition of terms to the reference datetime
+    """
+
+    # date and hour of the reference time
+    datetime: dt.datetime
+    # terms are time deltas vis-à-vis the reference input time step.
+    terms: np.array
+
+    # validity times are complete datetimes
+    validity_times: List[dt.datetime]
+
+
+GridConfig = namedtuple(
+    "GridConfig", "full_size latitude longitude geopotential landsea_mask"
+)
+
+
+@dataclass
+class Grid:
+    name: str
+    load_grid_info_func: Callable[
+        [Any], GridConfig
+    ]  # function to load grid data (customizable)
+    border_size: int = 10
+
+    # subdomain selection: If (0,0,0,0) the whole domain is kept.
+    subdomain: Tuple[int] = (0, 0, 0, 0)
+    # Note : training won't work with the full domain on some NN because the size
+    # can't be divided by 2. Minimal domain : [0,1776,0,2800]
+    x: int = field(init=False)  # X dimension of the grid (longitudes)
+    y: int = field(init=False)  # Y dimension of the grid (latitudes)
+    # projection information (e.g for plotting)
+    proj_name: str = "PlateCarree"
+    projection_kwargs: dict = field(default_factory={})
+
+    def __post_init__(self):
+        self.grid_config = self.get_grid_info()
+        # Setting correct subdomain if no subdomain is selected.
+        if sum(self.subdomain) == 0:
+            self.subdomain = (
+                0,
+                self.grid_config.full_size[0],
+                0,
+                self.grid_config.full_size[1],
+            )
+        self.x = self.subdomain[1] - self.subdomain[0]
+        self.y = self.subdomain[3] - self.subdomain[2]
+
+    def get_grid_info(self) -> GridConfig:
+        return self.load_grid_info_func(self.name)
+
+    @cached_property
+    def lat(self) -> np.array:
+        latitudes = self.grid_config.latitude[self.subdomain[0] : self.subdomain[1]]
+        return np.transpose(np.tile(latitudes, (self.y, 1)))
+
+    @cached_property
+    def lon(self) -> np.array:
+        longitudes = self.grid_config.longitude[self.subdomain[2] : self.subdomain[3]]
+        return np.tile(longitudes, (self.x, 1))
+
+    # TODO : from the grib, save a npy lat lon h mask for each grid
```

Comment:
i think we can remove this commented line

Reply:
obviously